docs/docsite/rst/os_guide/windows_winrm_kerberos.rst

```diff
@@ -0,0 +1,298 @@
+Delegation
+----------
+
+Kerberos delegation allows the credentials to traverse multiple hops. This is useful when you need to authenticate to a server and then have that server authenticate to another server on your behalf. To enable delegation, you must:
+
+* request a forwardable TGT when obtaining a ticket with ``kinit``
+* request the connection plugin to allow delegation to the server
+* the AD user is not marked as sensitive and cannot be delegated and is not a member of the ``Protected Users`` group
+* depending on the ``krb5.conf`` configuration, the target server may need to allow unconstrained delegation through its AD object delegation settings
+
+To request a forwardable TGT, either add the ``-f`` flag to the ``kinit`` command or set the ``forwardable = true`` option in the ``[libdefaults]`` section of the ``krb5.conf`` file. If you are using the ``psrp`` or ``winrm`` connection plugin to retrieve the TGT from the user's password in the inventory, it will automatically request a forwardable TGT if the connection plugin is configured to use delegation.
+
+To have the connection plugin delegate the credentials it will need to set the following hostvar in the inventory:
+
+.. code-block:: yaml+jinja
+
+    # psrp
+    ansible_psrp_negotiate_delegate: true
+
+    # winrm
+    ansible_winrm_kerberos_delegation: true
+
+    # ssh
+    ansible_ssh_common_args: -o GSSAPIDelegateCredentials=yes
+
+.. note::
+    It is also possible to set ``GSSAPIDelegateCredentials yes`` in the ``~/.ssh/config`` file to allow delegation for all SSH connections.
+
+To verify if a user is allowed to delegate their credentials, you can run the following PowerShell script on a Windows host in the same domain:
+
+.. code-block:: powershell
+
+    Function Test-IsDelegatable {
+        [CmdletBinding()]
+        param (
+            [Parameter(Mandatory)]
+            [string]
+            $UserName
+        )
+
+        $NOT_DELEGATED = 0x00100000
+
+        $searcher = [ADSISearcher]"(&(objectClass=user)(objectCategory=person)(sAMAccountName=$UserName))"
+        $res = $searcher.FindOne()
+        if (-not $res) {
+            Write-Error -Message "Failed to find user '$UserName'"
+        }
+        else {
+            $uac = $res.Properties.useraccountcontrol[0]
+            $memberOf = @($res.Properties.memberof)
+
+            $isSensitive = [bool]($uac -band $NOT_DELEGATED)
+            $isProtectedUser = [bool]($memberOf -like 'CN=Protected Users,*').Count
+
+            -not ($isSensitive -or $isProtectedUser)
+        }
+    }
+
+    Test-IsDelegatable -UserName username
+
+Newer versions of MIT Kerberos have added a configuration ok ``enforce_ok_as_delegate`` in the ``[libdefaults]`` section of the ``krb5.conf`` file. If this option is set to ``true`` delegation will only work if the target server account allows unconstrained delegation. To check or set unconstrained delegation on a Windows computer host, you can use the following PowerShell script:
```

Comment:
```suggestion
Newer versions of MIT Kerberos have added a configuration option ``enforce_ok_as_delegate`` in the ``[libdefaults]`` section of the ``krb5.conf`` file. If this option is set to ``true`` delegation will only work if the target server account allows unconstrained delegation. To check or set unconstrained delegation on a Windows computer host, you can use the following PowerShell script:
```

Reply:
not sure if "configuration ok" is what you meant here or not. just reads a little weird to me but I don't have a deep knowledge of Kerberos.

Reply:
Was a mistake, `option` is definitely better here.